Source/missiles.cpp

```diff
@@ -2686,202 +2698,167 @@ void AddDisarm(MissileStruct &missile, Point /*src*/, Point /*dst*/, int /*midir
 	}
 }
 
-void AddApoca(MissileStruct &missile, Point src, Point /*dst*/, int /*midir*/, int8_t /*mienemy*/, int id, int /*dam*/)
+void AddApoca(MissileStruct &missile, Point /*dst*/, int /*midir*/)
 {
 	missile._miVar1 = 8;
-	missile._miVar2 = src.y - missile._miVar1;
-	missile._miVar3 = missile._miVar1 + src.y;
-	missile._miVar4 = src.x - missile._miVar1;
-	missile._miVar5 = missile._miVar1 + src.x;
+	missile._miVar2 = std::max(missile.position.start.y - 8, 1);
+	missile._miVar3 = std::min(missile.position.start.y + 8, MAXDUNY - 1);
+	missile._miVar4 = std::max(missile.position.start.x - 8, 1);
+	missile._miVar5 = std::min(missile.position.start.x + 8, MAXDUNX - 1);

```

Comment:
Pre-existing bug but 1 and maxdun*-1 seem mismatched, should this be 0 and maxdun*-1? 1 and maxdun*-2?

Reply:
Yeah I noticed this as well, there is a similar issue elsewhere with the bound check in missiles, but it has no consequences for the game as the walls around the level are 16 tiles thick.